djangocms_navigation/admin.py

```diff
@@ -474,6 +519,58 @@ def changelist_view(self, request, menu_content_id=None, extra_context=None):
 
         return super().changelist_view(request, extra_context)
 
+    def set_main_navigation_view(self, request, menu_content_id):
+        """Sets the selected navigation as the main navigation, and unsets existing, 404 if invalid menu_content
+        or setting not enabled
+        :param: request: Request Object
+        :param: menu_content_id: Integer PK for menucontent
+        :return: redirect or 404
+        """
+        # Raise a 404 if the main_navigation functionality is disabled.
+        if not getattr(settings, "DJANGOCMS_NAVIGATION_MAIN_NAVIGATION_ENABLED", False):
+            raise Http404
+
+        changelist_url = get_admin_url(self.menu_content_model, "changelist")
+        menu_content = get_object_or_404(
+            self.menu_content_model._base_manager, id=menu_content_id
+        )
+        menu = menu_content.menu
+        menu_queryset = self.menu_model._base_manager.filter(
+            site=menu.site,
+            main_navigation=True,
+            menucontent__language=menu_content.language,
+        )
+        if request.method == "POST":
+            # If this is a POST method, then they have confirmed, change the main_navigation
+            menu_queryset.update(main_navigation=False)
+            menu.main_navigation = True
+            menu.save()
+            self.message_user(
+                request, _(f"You have set the navigation {menu.identifier} as the main navigation.")
+            )
+            return redirect(changelist_url)
+
+        # If this isn't a POST method, it is the first interaction, therefore redirect to confirmation
```

Comment:
```suggestion
        # If this isn't a POST method, it is the first interaction, therefore render confirmation
```
Very minor thing but you are not redirecting here so comment could cause confusion